packages/freetype.rb

```diff
@@ -62,5 +56,29 @@ def self.install
   def self.postinstall
     # make sure to delete downloaded files
     system "find #{CREW_BREW_DIR}/* -name freetype*.tar -exec rm -rf {} \+"
+    # This should become a function.
+    # check for conflicts with other installed files
+    @override_allowed = %w[fontconfig harfbuzz]
+    puts "Checking for conflicts with files from installed packages..."
+    conflicts = []
+    conflictscmd = %x[grep --exclude #{CREW_META_PATH}#{self.name}.filelist -Fxf #{CREW_META_PATH}#{self.name}.filelist #{CREW_META_PATH}*.filelist]
+    conflicts << conflictscmd.gsub(/(\.filelist|#{CREW_META_PATH})/, '').split("\n")
+    conflicts.reject!(&:empty?)
+    unless conflicts.empty?
+      if self.conflicts_ok?
+        puts "Handling conflict with the same file in another package.".orange
+      else
+        puts "Error: There is a conflict with the same file in another package.".lightred
+        @_errors = 1
+      end
+      conflicts.each do |conflict|
+        conflict.each do |thisconflict|
+          singleconflict = thisconflict.split(':',-1)
+          if @override_allowed.include?(singleconflict[0])
+            system "sed -i '\\\?^#{singleconflict[1]}?d'  #{CREW_META_PATH}/#{singleconflict[0]}.filelist"
+          end
+        end
+      end
+    end
```

Comment:
Should this logic replace the conflict handling code we have in crew?  Seems like it could be used universally?

Reply:
I think this function should ultimately be integrated into crew, yes.

But I was waiting on @supechicken 's work on making packages smaller in https://github.com/supechicken/chromebrew/tree/llvm_lib_diff to get done, but I haven't had time to go through that yet.

Reply:
Does this logic override what crew executes or does it run afterwards?  I'm guessing there are 2 conflicts checks running?

Reply:
But are there 2 checks or just the 1?  Ultimately, I think we should have just one check.

Reply:
There is an existing check at the moment, and this adds a second check in the post install.

As this second check isn't universal, the first (existing) check is still useful to tell us how the second check should be written. (I.e., we need to know which conflicting files, from which package, should be considered to be superseding.)